design/ceph/admission-controller.md

```diff
@@ -0,0 +1,93 @@
+
+The admission control process proceeds in two phases. In the first phase, mutating admission controllers are run. In the second phase, validating admission controllers are run. Note again that some of the controllers are both.
+
+If any of the controllers in either phase reject the request, the entire request is rejected immediately and an error is returned to the end-user.
+
+# Requirements
+
+1. Webhook server which will validate the requests
+2. TLS certificates for the server,
+3. ValidatingWebhookConfig/MutatingWebhookConfig which will intercept requests and send a HTTPS request to the webhook server.
+4. RBAC Components 
+
+#Admission Controllers in Rook
+
+![AC Image](./design-pic.png)
+
+##Setup
+###Certificate Management
+####Development and Testing 
+![AC Image](./ac_dev_design.png)
+
+We will be opting for using for self-signed certificates approved by the kubernetes cluster itself. Once common.yaml and the operator has been deployed.  Using scripts we will be performing the following tasks
+* Creating the private key and certs using openssl with the names tls.key and tls.crt. We will be hardcoding these names for now as it will be explained in later section
+* Creating and sending a CSR to kubernetes for approval
+* Once approved, a generic secret will be created with the given tls.key and tls.crt which will be later mounted onto the server pod for use.
+* Modifying the webhook.yaml to inject CA bundle onto the ValidatingWebhookConfig
+* Creating a webhook deployment, service, config on the cluster and deploying the entire webhook setup
+* Now our webhook server has been configured, deployed and ready to intercept requests based on the webhookconfig.
+* **All the above resources will be created in rook-ceph namespace hence it should be created in prior**
+
+Using the above approach, the dev/admins will have the responsibility of rotating the certificates when they expire.
+
+Below is an excerpt of what a ValidatingWebhookConfig looks like 
+
+```
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: demo-webhook
+webhooks:
+  - name: webhook-server.webhook-demo.svc
+    clientConfig:
+      service:
+        name: webhook-server
+        namespace: rook-ceph
+        path: "/validate"
+      caBundle: ${CA_PEM_B64}
+    rules:
+      - operations: [ "CREATE" ]
+        apiGroups: [""]
+        apiVersions: ["v1"]
+        resources: ["pods"]
+```
+
+We can make changes to the above values according to intercept based on whether a resource is being updated/deleted/created or change the type of resource or the request path which will be sent to the server.
+####Production
```

Comment:
nit: Some of the titles like this one are not rendered correctly. They need a blank line before and after the title, and a space between the hashes and the words

Reply:
Yes, i noticed images are not appearing properly. I have applied your changes (They need a blank line before and after the title, and a space between the hashes and the words). Not sure what is going wrong there.